spec/bson/hash_spec.rb

```diff
@@ -244,6 +244,16 @@ class HashSpecUnserializableClass
           /(Hash value for key 'foo'|Value) does not define its BSON serialized type:.*HashSpecUnserializableClass/)
       end
     end
+
+    context 'when hash is empty' do 
+      let(:buffer) do 
+        {}.to_bson
+      end
+
+      it 'has length 0' do 
```

Comment:
This test needs a different name. The situation we are testing is reading a buffer after having written to it previously, so this needs to be in the description and not the fact that hash length is zero when the hash is empty.